src/test/scala/ru/d4team/chat/services/ChatRoomsServiceSpec.scala

```diff
@@ -77,6 +82,32 @@ object ChatRoomsServiceSpec extends ZIOSpecDefault {
           _ <- chatRooms.leave(channel, room)
         } yield assertTrue(chatRooms.rooms == TrieMap.empty[String, ChatRoomMap])
 
+        testF.provide(ChatRoomsService.mapStorageLive)
+      }
+    },
+    test("'broadcast' should send a message from one channel to others in the same room") {
+      check(Gen.alphaNumericString, Gen.alphaNumericString, Gen.int(1, 10)) { (room, message, size) =>
+        val testF =
+          for {
+            // given
+            chatRooms     <- ZIO.service[ChatRoomsService[ChatRoomsMap]]
+            mainChannel   <- TestWebSocketChannel.make
+            mainChannels  <- List.fill(size)(TestWebSocketChannel.make).sequence
+            otherChannels <- List.fill(size)(TestWebSocketChannel.make).sequence
+            _             <- chatRooms.join(mainChannel, room)
+            _             <- ZIO.foreachParDiscard(mainChannels)(chatRooms.join(_, room))
+            _             <- ZIO.foreachParDiscard(otherChannels)(chatRooms.join(_, s"$room-2")) // add participants to another room
+            messageEvent   = ChannelRead(WebSocketFrame.text(message))
+
+            // then
+            _              <- chatRooms.broadcast(mainChannel, room, message)
+            channelEvents  <- mainChannel.asInstanceOf[TestWebSocketChannel].counterpartEvents.takeAll
```

Comment:
Don't like that `asInstanceOf[TestWebSocketChannel]`, but I don't get for now why `TestWebSocketChannel` is not subtype of `WebSocketChannel`
Anyway, we can avoid unsave operations by returing the `make` return type to `UIO[TestWebSocketChannel]` and just add an additional method `makeAsWebSocketChannel` for example with such code `def makeAsWebSocketChannel: UIO[WebSocketChannel] = make` and change initialization of channel in test to `make` if we need fields from `TestWebSocketChannel` and to 'makeAsWebSocketChannel' if we only need to assert

Reply:
A type parameter was added for `TestWebSocketChannel`'s `make ` method